gateway_code/open_nodes/tests/open_nodes_test.py

```diff
@@ -79,6 +79,22 @@ class MyNode(OpenNodeBase):
         ELF_TARGET = ('ELFCLASS32', 'EM_ARM')
         AUTOTEST_AVAILABLE = ['echo', 'get_time']
 
+        def setup(self, firmware_path):
```

Comment:
I don't think setup/teardown/status are needed. Just `verify`.

Reply:
no there is an error if you not put all abstract methods

Reply:
Ok, I had a look the previous code, and the check of `verify` should be done in another place. See https://github.com/iot-lab/iot-lab-gateway/blob/master/gateway_code/open_nodes/tests/open_nodes_test.py#L137

You can also subclass this class to check error cases.

Does it work when you leave the test_registry_open_node() unchanged ?

Reply:
Hi I will add an assert board_instance.verify() == 0 at the end of the file open_nodes_test.py
It doesn't work with test_registry_open_node() unchanged:
Error message: Can't instantiate abstract class MyNode with abstract methods setup, status, teardown

Reply:
It is not normal because this function doesn't instanciate MyNode. It just checks that the class is correctly registered.